app/src/marketplace/modules/communityProduct/services.js

```diff
@@ -67,10 +66,18 @@ export const createJoinPartStream = async (productId: ?ProductId = undefined): P
     try {
         const addEngineKeyPromises = [
             ...getStreamrEngineAddresses().map((address) => (
-                addStreamResourceKey(stream.id, address, 'share')
+                addPermission(stream.id, {
+                    operation: 'write',
+                    user: address,
+                })
             )),
+            // share permission is not strictly necessary but needed to an avoid error when
+            // removing user's share permission (must have at least one share permission)
             ...getStreamrEngineAddresses().map((address) => (
-                addStreamResourceKey(stream.id, address, 'write')
+                addPermission(stream.id, {
```

Comment:
This needed to be a permission, not a resource key. Only write permission would be necessary but share is needed to make the user's own share removal work.